src/rgl.cpp

```diff
@@ -1080,12 +1080,24 @@ void rglDisplayFramebuffers()
 
 void rglUpdate()
 {
-  int i;
-  
-  if (old_vi_origin == vi_origin) {
```

Comment:
The idea behind removing it was to replace it.  Sure, keeping it without removal and adding the additional exit rules would probably benefit speed, but it wouldn't fix any of the issues you mentioned, due to the `if (old_vi_origin == vi_origin)` case being premature information for single-buffered games, etc..

>I'm looking for a better place to call rglUpdate, maybe fullsync is a good place

I think from UpdateScreen-spawned functions is correct.

The RDP is not responsible for associating with the VI or screen refreshes.  If you chose to move rglUpdate() to within one of the RDP commands, such as FULLSYNC, then ROMs with no RDP command lists (like basic demos using the CPU and/or RSP to write to colored frame buffer) would never be rendered.